From 1ddb57c106ba5ea63bc15f612a2a1c3e29b8d6d4 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:24:32 -0400
Subject: [tools] perf/completion: Update __ltrim_colon_completions

Message-id: <1408462094-14194-104-git-send-email-jolsa@redhat.com>
Patchwork-id: 88121
O-Subject: [PATCH RHEL7.1 BZ1131394 103/325] perf completion: Update __ltrim_colon_completions
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 30079d1d5ebcb8d706c6e05cacebb7facc60cd95
Author: Ramkumar Ramachandra <artagnon@gmail.com>
Date: Thu Jul 4 18:11:26 2013 +0530

description
===========
The function is taken from the bash-completion package; update it to use
the latest version where colon_word doesn't miss quoting.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/bash_completion b/tools/perf/bash_completion
index 50540cf..b0cdd12 100644
--- a/tools/perf/bash_completion
+++ b/tools/perf/bash_completion
@@ -11,7 +11,7 @@ __ltrim_colon_completions()
 {
  if [[ "$1" == *:* && "$COMP_WORDBREAKS" == *:* ]]; then
   # Remove colon-word prefix from COMPREPLY items
-  local colon_word=${1%${1##*:}}
+  local colon_word=${1%"${1##*:}"}
   local i=${#COMPREPLY[*]}
   while [[ $((--i)) -ge 0 ]]; do
    COMPREPLY[$i]=${COMPREPLY[$i]#"$colon_word"}
-- 
1.7.1